pkg/build/builder/cmd/scmauth/sshkey.go

```diff
@@ -21,10 +23,33 @@ func (_ SSHPrivateKey) Setup(baseDir string, context SCMAuthContext) error {
 	if err := script.Chmod(0711); err != nil {
 		return err
 	}
+	foundPrivateKey := false
+	foundKnownHosts := false
+	files, err := ioutil.ReadDir(baseDir)
+	if err != nil {
+		return err
+	}
+	for _, file := range files {
+		switch {
+		case file.Name() == kownHostsFileName:
+			foundKnownHosts = true
+		case file.Name() == SSHPrivateKeyMethodName:
+			foundPrivateKey = true
+		}
+		log.V(5).Infof("source secret dir %s has file %s", baseDir, file.Name())
+	}
+	if !foundPrivateKey {
+		return fmt.Errorf("could not find the ssh-privatekey file for the ssh secret stored at %s", baseDir)
+	}
+	// let's see if known_hosts was included in the secret
+	knownHostsPath := filepath.Join(baseDir, kownHostsFileName)
 	content := "#!/bin/sh\nssh -i " +
```

Comment:
I have the impression that now this `-i` is lost as it does not contain, as flag value, the path for the private key. The command become something like:

```
ssh -i -o StrictHostKeyChecking=false $@
```

Reply:
yep I inadvertently dropped the filename ... will address - thanks